packages/mobile/src/store/offline-downloads/slice.ts

```diff
@@ -188,43 +193,83 @@ const slice = createSlice({
         delete state.collectionStatus[collectionId]
       })
     },
-    loadTracks: (state, { payload: tracks }: PayloadAction<LineupTrack[]>) => {
-      tracks.forEach((track) => {
-        const trackIdStr = track.track_id.toString()
-        state.tracks[trackIdStr] = track
-        state.downloadStatus[trackIdStr] = OfflineDownloadStatus.SUCCESS
+    batchSetTrackOfflineMetadata: (
+      state,
+      action: PayloadAction<TrackOfflineMetadataPayload[]>
+    ) => {
+      const { payload: trackOfflineMetadatas } = action
+      trackOfflineMetadatas.forEach((trackOfflineMetadata) => {
+        const { trackId, offlineMetadata } = trackOfflineMetadata
+        state.offlineTrackMetadata[trackId] = offlineMetadata
+      })
+    },
+    batchAddTrackDownloadReason: (
```

Comment:
@dylanjeffers these are placeholders as we currently use `batchSetTrackOfflineMetadata`, but I know you're waiting on these for your changes. Maybe we can sync on the best structure

Reply:
sweet, yeah i like batchSetTrackOfflineMetadata

Reply:
do you want me to remove the others? we don't use the add/remove download reason anymore in my change